Downgrade .NET framework dependency to 4.5 version

y-skindersky opened the pull request:
It would be nice to have 4.5 version for further usage in ZeroLog, for example.

y-skindersky committed:
[*] Downgrade to 4.5 framework dependency

y-skindersky committed:
[*] Downgrade to 4.5 framework dependency

y-skindersky committed:
[*] Downgrade to 4.5 framework dependency

MendelMonteiro commented:
The tests fail when running the cake build script.  I get a "Could not load file or assembly 'System.Runtime, Version=4.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a'" exception.

Could you check this out?

y-skindersky commented:
I thought I solved it by referencing NETStandard.Library v1.6.0. I had the same problem and after adding this package all the tests are running well under cake.

MendelMonteiro commented:
I've checked out your fork and apparently it's still an issue.

y-skindersky commented:
It seems StringFormatter uses tuples without referencing System.ValueTuple package. I have VS2017 15.5.7 and 4.7 framework installed and don't see any errors in VS while on another machine with earlier version of VS and framework build fails. So we need to add missed reference and to investigate this strange VS behavior (maybe [this link](https://developercommunity.visualstudio.com/content/problem/95070/could-not-load-file-or-assembly-systemruntime-vers.html#reply-98340) could be helpful)

y-skindersky committed:
[-] Disable AutoGenerateBindingRedirects

y-skindersky commented:
Please try updated version with disabled AutoGenerateBindingRedirects flag. We also need to add reference to System.ValueTuple for the case if any contributor has not the latest framework version installed.

y-skindersky committed:
Add explicit reference to System.ValueTuple.

This package has become a built-in member of .NET Framework 4.6.2 and .NET Core/Standard 2.0. Since project is targeting lower versions of framework explicit reference is required. Otherwise the project won't build if only .NET 4.5 is installed.

MendelMonteiro commented:
A new Nuget package has been uploaded
